### PR TITLE
cleaned up BE

### PR DESCRIPTION
**Pull request by Lymetta**
Did the code refactoring according to mentor's recommendations. Please check if I missed anything. I used Big Decimal for calculation results only (I doubt user or admin will send big decimal type input), didn't add @Qualifier (still discussing with mentor, will update if anything changes).

### CODE REVIEW

**Review by AudriusAnusauskas — COMMENTED** [review]
I think everything seems fine, changed the way agreed yesterday. At least I did not find anything missing.
The naming of the data fields should be agreed. As I see you have classes for AdminData and CalculatorData. And fields that are in common for admin you simply add 'admin' in the beginning of each field name. And that is a good idea. So I think there should be a consistency in naming. Like if you have 'euriborInterestRate' and 'adminEuriborInterest' I think they should be similar, just with and without prefix 'admin'. Also with min, max and default 'price' in admin and 'value' in calculator. I think they should be same, just difference in prefix.
Just my thoughts :)

File: server/src/main/java/com/server/loan/calculator/controllers/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.server.loan.calculator.models.AdminData;
+import com.server.loan.calculator.services.AdminService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("admin")
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/controllers/AdminController.java:11]: i would suggest putting '/' before the endpoint name. i.e. @ RequestMapping("/admin")

File: server/src/main/java/com/server/loan/calculator/controllers/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@RequiredArgsConstructor
+public class AdminController {
+
+    private final AdminService adminService;
+
+    @GetMapping("values")
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/controllers/AdminController.java:17]: same as above, but also make the endpoint name more meaningful like "/get-values"

File: server/src/main/java/com/server/loan/calculator/controllers/CalculatorController.java [hunk 3]
```diff
@@ -1,24 +1,20 @@
-package com.server.loanCalculator.calculator;
+package com.server.loan.calculator.controllers;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/controllers/CalculatorController.java:1]: package name should be singular - controller

File: server/src/main/java/com/server/loan/calculator/controllers/CalculatorController.java [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
-@Slf4j
 @RestController
-@RequestMapping("")
+@RequestMapping("/calc")
 @RequiredArgsConstructor
 public class CalculatorController {
 
     private final CalculatorService calculator;
 
     @GetMapping
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/controllers/CalculatorController.java:16]: add a meaningful name to this endpoint, like "calculate-mortgage"

File: server/src/main/java/com/server/loan/calculator/errors/ErrorMessage.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.server.loanCalculator.errors;
+package com.server.loan.calculator.errors;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/errors/ErrorMessage.java:1]: package name should be singular - error

File: server/src/main/java/com/server/loan/calculator/services/AdminService.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+@Service
+@Data
+public class AdminService {
+
+    private final AdminData adminData;
+    private final ValueInRangeValidator valueInRangeValidator;
+    private final InputAsCorrectNumberValidator inputValidator;
+
+    private void validateAdminInput () {
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/AdminService.java:19]: would make more sense to move this validation to a separate validator class like DataValidator

File: server/src/main/java/com/server/loan/calculator/services/CalculatorService.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
+package com.server.loan.calculator.services;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/CalculatorService.java:1]: package name should be singular - service

File: server/src/main/java/com/server/loan/calculator/services/CalculatorService.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import com.server.loan.calculator.validators.InputAsCorrectNumberValidator;
+import com.server.loan.calculator.validators.ValueInRangeValidator;
+import lombok.Data;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+@Service
+@Data
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/CalculatorService.java:16]: service should not have @ Data annotation

File: server/src/main/java/com/server/loan/calculator/services/CalculatorService.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+public class CalculatorService {
+
+    private final CalculatorData data;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/CalculatorService.java:19]: should not be auto injected, later you will instantiate this class through endpoint controller or database DAO, so in this case you should just do new CalculatorData()

File: server/src/main/java/com/server/loan/calculator/models/AdminData.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package com.server.loan.calculator.models;
+
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
+public class AdminData {
+    private double adminEuriborInterest = 3.31;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/models/AdminData.java]: all money related variables should be BigDecimal

File: server/src/main/java/com/server/loan/calculator/models/CalculatorData.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package com.server.loan.calculator.models;
+
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
+public class CalculatorData {
+    private double euriborInterestRate = 3.31;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/models/CalculatorData.java]: all money related variables should be BigDecimal

File: server/src/main/java/com/server/loan/calculator/services/CalculatorService.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    private final ValueInRangeValidator valueInRangeValidator;
+    private final InputAsCorrectNumberValidator inputValidator;
+
+    private double calculateMonthlyInterestRateDecimals () {
+        return ((data.getEuriborInterestRate() + data.getBankInterestMargin()) / 100) / 12;
+    }
+
+    private BigDecimal calculateMaxAvailableLoan () {
+        inputValidator.validate(data.getSalary());
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/CalculatorService.java]: all this validation should be moved to a separate class, and should be called before calling calculation methods

File: server/src/main/java/com/server/loan/calculator/services/CalculatorService.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+        inputValidator.validate(data.getMortgagePeriodYears());
+        valueInRangeValidator.validate(data.getMortgagePeriodYears(), 1, 30);
+        double monthlyAvailableMoney = (data.getSalary() * 0.4) - (data.getFinancialObligations());
+        BigDecimal mathPower = BigDecimal.valueOf(Math.pow(1 + (calculateMonthlyInterestRateDecimals()), -(12 * data.getMortgagePeriodYears())));
+        return (BigDecimal.valueOf(monthlyAvailableMoney)
+                .multiply((BigDecimal.ONE.subtract(mathPower)))).divide(BigDecimal.valueOf(calculateMonthlyInterestRateDecimals()), RoundingMode.HALF_UP);
+    }
+
+    private BigDecimal calculateMonthlyPayment () {
+        valueInRangeValidator.validate(data.getPropertyValue(), data.getMinPropertyValue(), data.getMaxPropertyValue());
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/services/CalculatorService.java]: all this validation should be moved to a separate class, and should be called before calling calculation methods

File: server/src/main/java/com/server/loan/calculator/validators/InputAsCorrectNumberValidator.java [hunk 14]
```diff
@@ -0,0 +1,14 @@
+package com.server.loan.calculator.validators;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/validators/InputAsCorrectNumberValidator.java]: package name should be singular - validator

File: server/src/test/java/com/server/loan/LoanCalculatorApplicationTests.java [hunk 15]
```diff
@@ -1,4 +1,4 @@
-package com.server.loanCalculator;
+package com.server.loan;
```
Comment (Rytautas) [inline server/src/test/java/com/server/loan/LoanCalculatorApplicationTests.java:1]: should be moved to com.server.loan.calculator

**Review by Rytautas — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/server/loan/calculator/service/AdminService.java [hunk 16]
```diff
@@ -0,0 +1,17 @@
+package com.server.loan.calculator.service;
+
+import com.server.loan.calculator.model.AdminData;
+import com.server.loan.calculator.validator.AdminDataValidator;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AdminService {
+
+    private final AdminData adminData = new AdminData();
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/service/AdminService.java:10]: would make more sense to instantiate this inside the method that you are using to return it, as later you will be getting it from database

File: server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+import com.server.loan.calculator.exception.ValidationException;
+import com.server.loan.calculator.model.AdminData;
+import io.micrometer.common.util.StringUtils;
+import lombok.RequiredArgsConstructor;
+
+import java.math.BigDecimal;
+
+@RequiredArgsConstructor
+public class AdminDataValidator {
+    private final AdminData adminData;
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java:12]: should not be here as a field. methods that use/validate data should take it as a parameter

File: server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    private final BigDecimalValueInRangeValidator valueInRangeValidator = new BigDecimalValueInRangeValidator();
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java:13]: with @ RequiredArgsConstructor annotation you dont need to instantiate these classes, if they are annotated with @ Component or @ Service annotations, which they should be

File: server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    private final BigDecimalInputAsCorrectNumberValidator inputValidator = new BigDecimalInputAsCorrectNumberValidator();
```
Comment (Rytautas) [inline server/src/main/java/com/server/loan/calculator/validator/AdminDataValidator.java:14]: with @ RequiredArgsConstructor annotation you dont need to instantiate these classes, if they are annotated with @ Component or @ Service annotations, which they should be

**Pull request merged**